google_guest_agent/network/manager/common.go

```diff
@@ -0,0 +1,73 @@
+//  Copyright 2024 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package manager is responsible for detecting the current network manager service, and
+// writing and rolling back appropriate configurations for each network manager service.
+package manager
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+var (
+	badMAC = make(map[string]net.Interface)
+)
+
+// InterfaceListsIpv4Ipv6 gets a list of interface names. The first list is a list of all
+// non-primary interfaces, and the second list consists of only non-primary interfaces
```

Comment:
Is this comment accurate about returning only secondary interfaces? Looks like code is iterating through all interfaces and not really skipping primary one

Reply:
Yea, I forgot to update the comment when I updated the method.